rust/otap-dataflow/crates/controller/src/lib.rs

```diff
@@ -94,13 +145,20 @@ impl<PData: 'static + Clone + Send + Sync + std::fmt::Debug> Controller<PData> {
                 }
                 Err(e) => {
                     // Thread join failed, handle the error
-                    return Err(error::Error::InternalError {
-                        message: format!("Failed to join thread pipeline-core-{core_id}: {e:?}"),
+                    return Err(error::Error::ThreadPanic {
```

Comment:
Not related to changes in this PR but we are not waiting for all threads to join. The very first thread that returns `Errr(e)` or panics, would exit the loop and return. We should collect all possible errors and join all threads before returning.

Reply:
Add a ToDo in the code, I will most likely work on the controller next anyway.